iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1265,6 +1277,19 @@ static int setMqttMessagePropertyIfPossible(IOTHUB_MESSAGE_HANDLE IoTHubMessage,
     // Not finding a system property to map to isn't an error.
     int result = 0;
 
+    if (nameLen > 5)
+    {
+        if (strcmp((const char*)&propName[nameLen - 5], MESSAGE_CREATION_TIME_UTC) == 0)
```

Comment:
same here what does `5` represent

Reply:
This is how it's implemented in the rest of this function. Would require refactoring the function to make this not a one off.